packages/near-api-js/src/constants.ts

```diff
@@ -1,9 +1,17 @@
 import BN from 'bn.js';
 
-// Default amount of gas to be sent with the function calls. Used to pay for the fees
-// incurred while running the contract execution. The unused amount will be refunded back to
-// the originator.
-// Due to protocol changes that charge upfront for the maximum possible gas price inflation due to
-// full blocks, the price of max_prepaid_gas is decreased to `300 * 10**12`.
-// For discussion see https://github.com/nearprotocol/NEPs/issues/67
-export const DEFAULT_FUNCTION_CALL_GAS = new BN('30000000000000');
\ No newline at end of file
+/** Default amount of gas to be sent with the function calls. Used to pay for the fees
+* incurred while running the contract execution. The unused amount will be refunded back to
+* the originator.
+* Due to protocol changes that charge upfront for the maximum possible gas price inflation due to
+* full blocks, the price of max_prepaid_gas is decreased to `300 * 10**12`.
+* For discussion see https://github.com/nearprotocol/NEPs/issues/67
+*/
+export const DEFAULT_FUNCTION_CALL_GAS = new BN('30000000000000');
+
+/**
+ * Hash if no contract is present for an account.
+ */
+export const EMPTY_CONTRACT_HASH = '11111111111111111111111111111111';
+
+export const ZERO_NEAR = new BN('0');
```

Comment:
@willemneal do we need this? `Account.functionCall` does not have a fallback for `attachedDeposit` can you describe the failure mode you're trying to avoid here?

Reply:
Do you mean the ZERO_NEAR? I'm a little confused with the context.  Why did the default for function call have zero attached deposit?  If that's the question it's because that is the common case. Not really for any failure, just convenience.

Reply:
Yes I'm wondering why we aren't doing the same for function call actions sent from `Account.functionCall`. Maybe the fallback is handled at the protocol level. Is there a difference between a function call action with no deposit specified and one with `0`? `Account.functionCall` actions don't specify a deposit when none is passed in currently.

Reply:
Are these changes relevant to `TransactionBuilder` at all? If not could they go in a separate PR please?

Reply:
Yeah `TransactionBuilder` function call actions are defaulting to `ZERO_NEAR` when no deposit is passed in

Reply:
I'm removing these features incl the fallback and `hasDeployedContract` for now. We can discuss if they're necessary in a separate PR / issue